yt/frontends/gamer/cfields.pyx

```diff
@@ -0,0 +1,215 @@
+# distutils: include_dirs = LIB_DIR
+# distutils: libraries = STD_LIBS
+cimport cython
+cimport libc.math as math
+cimport numpy as np
+import numpy as np
+
+np.import_array()
+
+cdef np.float64_t h_eos4(np.float64_t e):
+    cdef np.float64_t x
+    x = 2.25 * e * e
+    return 2.5 * e + x / (1.0 + math.sqrt(x + 1.0))
+
+cdef np.float64_t _lorentz_factor_eos4(
+        np.float64_t rho, 
+        np.float64_t mx,
+        np.float64_t my,
+        np.float64_t mz,
+        np.float64_t e,
+        np.float64_t c
+    ):
+    cdef np.float64_t u2, c2, vx, vy, vz
+    cdef np.float64_t htilde, fac;
+
+    c2 = c * c
+    fac = (1.0 / (rho * (h_eos4(e) + 1.0))) ** 2
+    u2 = (mx * mx + my * my + mz * mz) * fac
+    return math.sqrt(1.0 + u2 / c2)
+
+def gamma_eos4(temp):
+    cdef np.float64_t[:] kT = temp.ravel()
+    
+    out = np.empty_like(kT)
+    cdef np.float64_t[:] outp = out.ravel()
+    cdef np.float64_t x, c_p, c_v
+    cdef int i
+
+    for i in range(outp.shape[0]):
+        x = 2.25 * kT[i] / math.sqrt(2.25 * kT[i] * kT[i] + 1.0)
+        c_p = 2.5 + x
+        c_v = 1.5 + x
+        outp[i] = c_p / c_v
+    return out
+
+cdef np.float64_t cs_eos4(np.float64_t kT, np.float64_t c):
+    cdef np.float64_t hp, cs2;
+    hp = h_eos4(kT) + 1.0
+    cs2 = kT / (3.0 * hp)
+    cs2 *= (5.0 * hp - 8.0 * kT) / (hp - kT)
+    return c * math.sqrt(cs2)
+
+def sound_speed_eos4(temp, np.float64_t c):
+    cdef np.float64_t[:] kT = temp.ravel()
+    out = np.empty_like(kT)
+    cdef np.float64_t[:] outp = out.ravel()
+    cdef np.float64_t hp, cs2;
+
+    cdef int i
+    for i in range(outp.shape[0]):
+        outp[i] = cs_eos4(kT[i], c)
+    return out
+
+cdef np.float64_t _four_vel(
```

Comment:
```suggestion
cdef inline np.float64_t _four_vel(
```

Reply:
I checked and it has no impact on performance.

Reply:
OK! Thanks for checking.